src/engine/shared/console.h

```diff
@@ -15,8 +15,11 @@ class CConsole : public IConsole
 {
 	class CCommand : public CCommandInfo
 	{
-	public:
 		CCommand *m_pNext;
+
+	public:
+		CCommand *Next() const { return m_pNext; }
```

Comment:
This is a const getter returning a non-const pointer. See discussion in #10868. I think what this discussion missed is that this approach makes more sense when there is non-trivial work or validation being performed in the getter that should not be duplicated. I think in trivial cases like this, the implementation `return m_pNext;` could simply be duplicated in the const and non-const getters instead of doing the double `const_cast` dance.

```suggestion
const CCommand *Next() const { return m_pNext; }
CCommand *Next() { return m_pNext; }
```

OR

```suggestion
const CCommand *Next() const { return m_pNext; }
CCommand *Next() { return const_cast<CCommand *>(const_cast<const CCommand *>(this)->Next()); }
```

Reply:
applied